Use constant id for debug output in legacy RequestConnection

gavin-norman-sociomantic opened the pull request:
The object_pool_index that is currently used can change, when items are
added or removed from the object pool. This is very confusing during
debugging!

Fixes #148.

nemanja-boric-sociomantic commented on src/swarm/client/connection/RequestConnection.d:
```diff
@@ -364,6 +382,8 @@ public abstract class IRequestConnection :
         assert(conn_pool !is null, typeof(this).stringof ~ ".ctor: connection pool interface must be non-null");
         assert(params !is null, typeof(this).stringof ~ ".ctor: request params instance must be non-null");
 
+        this.id = instance_count++;
```
`this.id = typeof(this).instance_count++` or similar? Do we require `this.` on the static members?

gavin-norman-sociomantic replied:
Updated.

Commit:
Use constant id for debug output in legacy RequestConnection

The object_pool_index that is currently used can change, when items are
added or removed from the object pool. This is very confusing during
debugging!

Fixes #148.
